api/scpca_portal/models/project.py

```diff
@@ -290,13 +241,15 @@ def combine_multiplexed_metadata(
         if not multiplexed_libraries_metadata:
             return combined_metadata, multiplexed_sample_mapping
 
-        # Get all the field names to pass to the csv.DictWriter
-        all_fields = set(multiplexed_libraries_metadata[0].keys())
-        all_fields.update(
-            set(samples_metadata[0].keys())
-            - set(self.output_multiplexed_metadata_ignored_fields["injected"])
-            - set(self.output_multiplexed_metadata_ignored_fields["single_cell"])
+        modalities = (Sample.MULTIPLEXED_MODALITY,)
+        ignored_library_metadata_keys = self.get_ignored_library_metadata_keys(
+            modalities=modalities
         )
+        ignored_sample_metadata_keys = self.get_ignored_sample_metadata_keys(modalities=modalities)
+
+        # Get all the field names to pass to the csv.DictWriter
+        all_fields = set(multiplexed_libraries_metadata[0].keys()) - ignored_library_metadata_keys
+        all_fields.update(set(samples_metadata[0].keys()) - ignored_sample_metadata_keys)
```

Comment:
I can't comment below this line but around 255 I think we should iterate over the keys in the metadata that exist and sort them based on the list of `output_multiplexed_metadata_field_order`.

Reply:
Reworked based on your comment.